### PR TITLE
feat(mobile): notify mobile layer of thread updates

### PR DESCRIPTION
**Pull request by sanderpick**
also fixes some goroutine in loop bugs

### CODE REVIEW

File: mobile/mobile.go [hunk 1]
```diff
@@ -18,41 +18,54 @@ import (
...
 	var m Mobile
-	return m.NewNode(repoPath, centralApiURL, logLevel)
+	return m.NewNode(config, messenger)
 }
 
 // Mobile is the name of the framework (must match package name)
 type Mobile struct{}
 
 // Create a gomobile compatible wrapper around TextileNode
-func (m *Mobile) NewNode(repoPath string, centralApiURL string, logLevel string) (*Wrapper, error) {
```
Comment (asutula) [inline mobile/mobile.go]: So how is log level working now?

File: mobile/mobile.go [hunk 2]
```diff
@@ -246,10 +257,29 @@ func (w *Wrapper) PairDesktop(pkb64 string) (string, error) {
...
+	datac := make(chan tcore.ThreadUpdate)
+	go tcore.Node.JoinRoom(id, datac)
+	go func() {
+		for {
+			select {
+			case update, ok := <-datac:
+				if !ok {
+					return
+				}
+				w.messenger.Notify(&update)
```
[the remark this answers is not in the document]
Comment (asutula) [reply mobile/mobile.go]: awesome

File: mobile/mobile.go (hunk 1 above)
Comment (sanderpick) [reply mobile/mobile.go]: It's the same but you set `LogLevel` on the config object instead. There were getting to be many constructor params.

**Comment by sanderpick** [comment]
Refactored this to take a generic `Event` struct, not `core.ThreadUpdate`, which means a) mobile frameworks don't need `core` package, and `Messenger` can be used for any event. The bridges just pass on event name and the un-serialized payload object (`NSDictionary` / `WritableMap`)

**Comment by sanderpick** [comment]
okay guys, this is g2g

**Pull request merged**